extension/observer/k8sobserver/factory.go

```diff
@@ -35,10 +35,10 @@ const (
 
 // Factory is the factory for the extension.
 type Factory struct {
-	createK8sConfig func() (*rest.Config, error)
+	createK8sClientset func(config k8sconfig.APIConfig) (*kubernetes.Clientset, error)
```

Comment:
Why having this func as a param in the struct vs a function on the factory itself? If it is because of testing please document/comment in the code.

Reply:
I believe that may have been the original motivation. @jrcamp can you confirm?

Reply:
Then just add a comment, because I see it used in the test for the moment.

Reply:
Done.

Reply:
Yes it's used for testing. It might be able to be restructured now that there's kuberntes.APIConfig and for tests use AuthTypeNone but if it's not obvious how to I'm ok leaving it how it is.